File: HahnTest.Server/Entities/Employee.cs
using System.Xml.Serialization;

namespace HahnTest.Server.Entities
{
    /// <summary>
    /// For this exercise, XML decorators to mapo the entities easier
    /// </summary>
    public class Employee
    {
        [XmlElement("Id")]
        public string Id { get; set; }

        [XmlElement("Name")]
        public string Name { get; set; }

        [XmlElement("Position")]
        public string Position { get; set; }

        [XmlElement("Salary")]
        public decimal Salary { get; set; }
    }
}

File: hahntest.client/src/types/employee.ts

/**
 * 
 * @export
 * @interface Employee
 */
export interface Employee {
    /**
     * 
     * @type {string}
     * @memberof Employee
     */
    'id': string ;
    /**
     * 
     * @type {string}
     * @memberof Employee
     */
    'name'?: string | null;
    /**
     * 
     * @type {string}
     * @memberof Employee
     */
    'position'?: string | null;
    /**
     * 
     * @type {number}
     * @memberof Employee
     */
    'salary'?: number;
}